pkg/logs/hclog.go

```diff
@@ -0,0 +1,184 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"io"
+	"log"
+
+	"github.com/hashicorp/go-hclog"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// NewHCLogAdapter creates a new adapter, wrapping an underlying
+// zap.Logger inside an implementation that emulates hclog.Logger.
+func NewHCLogAdapter(wrapped *zap.Logger) *HCLogAdapter {
+	if wrapped == nil {
+		wrapped = zap.L()
+	}
+	wrapped = zap.New(wrapped.Core(), zap.AddCallerSkip(1))
+	return &HCLogAdapter{
+		name: "",
+		zap:  wrapped,
+	}
+}
+
+// HCLogAdapter is an adapter that allows to use a zap.Logger where
+// and hclog.Logger is expected.
+type HCLogAdapter struct {
+	name string
+	zap  *zap.Logger
+}
+
+// This variable is a guard to ensure that HCLogAdapter actually satisfies the hclog.Logger interface
+var _ hclog.Logger = (*HCLogAdapter)(nil)
+
+func (l *HCLogAdapter) Clone() *HCLogAdapter {
+	return &HCLogAdapter{
+		name: l.name,
+		zap:  l.zap,
+	}
+}
+
+func (l *HCLogAdapter) Log(level hclog.Level, msg string, args ...interface{}) {
+	var fields []zapcore.Field
+	for i := 0; i < len(args); i += 2 {
+		fields = append(fields, zap.Any(args[i].(string), args[i+1]))
+	}
```

Comment:
Extract this into a function and maybe do some range checking to panic with some nicer message?